src/actions/ProductsActions.ts

```diff
@@ -57,3 +69,23 @@ export const getProductsByProvider = (id: string) => {
       });
   });
 };
+
+export const getProductById = (id: string) => {
+  return new Promise<product>((resolve, reject) => {
+    useQuery<getProductByIdReturnTypes>(getProductByIdQuery, { id }).then(
```

Comment:
What about error?

Reply:
For the moment I'm handling two types of errors.

#### 1: Error connecting with the database (Network or another like server, etc...)

This error is handled by the `useQuery` util function

https://github.com/Azordev/did-app/blob/ee63cc65a23ddbbc6ad7c451efe0189c17279c81/src/utils/apollo.ts#L49-L56

#### 2: You're connected correctly to the database but that you're searching doesn't exits (ID invalid, etc...)

This is handled individually by each action, this, for example, is what this action does when not found any product:

https://github.com/Azordev/did-app/blob/9b26fec004737e6ec06066510401093be0a107cb/src/actions/ProductsActions.ts#L54-L62

Reply:
Mmm I've checked and I need to redirect the user when a query fails jeje 

![LastRecord](https://user-images.githubusercontent.com/66505715/171781163-163f5bb6-eed8-4e7c-8145-e6697524a4e2.gif)

Reply:
I've missed too to add the logger, thanks for notifying me!

Reply:
I've added the logger now. I think we can think of a way to handle errors for a new issue, I'll create the discussion

Reply:
Remember there is a "frontend-errors" table on Hasura DB to send errors in production

Reply:
I've checked and I've noticed that:

1. When the query fails is shown a notification
2. When the query does not fail but the result is empty is shown a notification (in some cases)

But the user just sees the notification and gets on an empty view. I think the user should be also redirected.

I've created a new issue to create this functionality on all the views that require it

https://github.com/Azordev/did-app/issues/38

Reply:
So for now if any error happens, we need to show a notification (as it happening), then if it is in production send the error to Hasura.

Reply:
> Remember there is a "frontend-errors" table on Hasura DB to send errors in production

Oh, the logger util function has that idea, but still is not integrated with the Hasura DB. I'll create the issue to work that logic